### ISSUE TITLE
Spurious Oauth Access 

### ISSUE THREAD

**Issue by RoosterSixOne** [k0shk0sh/FastHub#2224]
10 minutes ago, I received a notification that this application was authorized to access my github account. I did not request this. In fact, this github account has remained dormant for a very long time. Why was FastHub given access to my account?

The actor reported in the github audit trail entry from when oauth access was granted is apparently from Guatemala at IP 

 181.174.102.168

**Comment by rurikid** [comment]
Same with me. Security logs show a long history of attempted access to my account followed by enabling FastHub Oauth.

**Comment by k0shk0sh** [comment]
Please follow on this: https://www.reddit.com/r/github/comments/aglt3z/i_just_had_fasthub_randomly_authorized_on_my/

**Comment by RoosterSixOne** [comment]
Please do not close the issue and shunt the discussion to reddit. It makes the problem less visible, and gives the appearance that this issue has been solved, which it has not. 

An outside entity is clearly using FastHub as an attack vector in some fashion, and until the problem has been investigated and mitigated, it should remain here. Reddit threads quickly fade from sight.

**Comment by RoosterSixOne** [comment]
Certainly. 
![githubscreenshot](https://user-images.githubusercontent.com/17522042/51270559-416c1500-1993-11e9-9573-263eea39cade.png)

**Comment by rurikid** [comment]
![screenshot from 2019-01-16 12-30-31](https://user-images.githubusercontent.com/30425208/51270595-f9e18b00-198a-11e9-9761-d2019ded8a06.png)
![screenshot from 2019-01-16 12-32-38](https://user-images.githubusercontent.com/30425208/51270598-fc43e500-198a-11e9-9670-f262aee3c4d3.png)

**Comment by RoosterSixOne** [comment]
I'd blur your creds there rurikid. You don't know whose looking at this page.

**Comment by TDashDog1** [comment]
+1 to this, this happened to me (I don't even use this account anymore)

**Comment by RoosterSixOne** [comment]
Already done, @dedepete. This is an old account and was created back before 2FA was a widespread thing for github.

**Comment by k0shk0sh** [comment]
It can definitely be breached by decompiling the app or listening to the traffic.

**Comment by k0shk0sh** [comment]
I follow up with another two emails to GitHub, one to point to this thread and one to conclude that the accounts that are being hacked are old & inactive as it seem that everyone who had this reported that their account is years old and never been used. 

Also guys, the hacker might just used the name FastHub but not the actual token  of FastHub. 

Rest assured, FastHub (me) never take data outside of the app. That's something I can guarantee with my life :)

**Comment by k0shk0sh** [comment]
So far GitHub didn't reply to any of my emails anymore! Perfect service I recon as usual! 

Did they reply with anything guys?

**Comment by dedepete** [comment]
@k0shk0sh so what can we do to prevent this in future?

**Comment by k0shk0sh** [comment]
@dedepete there is nothing really that we can do unfortunately, even if I encrypt the token, still someone can listen to traffic via Charles proxy and get it, or even create a token and give it a FastHub name and it should look like it was from FastHub, I believe that this attack was mainly to somehow give FastHub bad image otherwise honestly why would they use that name and not other GitHub apps such as octodroid for example!. 

As soon as I finish up v5 I'll be taking seriously any kind of suspicious fork as a copyright issue or something like this. 

The community unfortunately some of them doesn't get what other people has done to get their source code to be as famous as FH for example, the time I spent on this app after working hours, days and nights during hard times or happy times, I just don't get why people would behave so bad on something that was done to serve them!. 

I honestly everyday I see something that demotivate me from continuing evolving this app.

**Comment by paul300497** [comment]
Whoever did this, might have not listened to traffic. It was smart, the accounts were monitored for some time to see the activity, and so the unused accounts were hijacked. It is foolproof, the name, the scheme, it all. Who gets active again, gets their account back, who doesn't, well they have their passwords changed and so the hijacker has more accounts. The mistake was that once I gave authorization (not me, but the person who did this), right after my pass was changed. If I wouldn't have done anything, it would have stood changed. If I would have done something, which obviously I did, I get back my account
